### DISCUSSION TITLE
What is the best way to extend Pimcore's core features?

### DISCUSSION THREAD

**Discussion by ghost** [pimcore/pimcore#16987] [Q&A / Support]
Core functionality should be expanded from time to time. What is the best way to proceed? Botching directly in the vendor code is ... bad :)

**Comment by markus-moser** [comment]
This depends on what you would like to customize, but you would need to create and enable a Pimcore bundle with additional PHP or JS code. 

https://pimcore.com/docs/platform/Pimcore/Extending_Pimcore/Bundle_Developers_Guide/Pimcore_Bundles/
